code/modules/clothing/spacesuits/rig.dm

```diff
@@ -97,639 +111,310 @@
 	allowed = list(/obj/item/device/flashlight,/obj/item/weapon/tank,/obj/item/weapon/storage/bag/ore,/obj/item/device/t_scanner,/obj/item/weapon/pickaxe, /obj/item/device/rcd, /obj/item/weapon/wrench/socket)
 	max_heat_protection_temperature = SPACE_SUIT_MAX_HEAT_PROTECTION_TEMPERATURE
 	pressure_resistance = 200 * ONE_ATMOSPHERE
+	actions_types = list(/datum/action/item_action/toggle_rig_suit)
+
+	var/activated = FALSE
+	var/list/modules = list()
+	var/list/initial_modules = list(/obj/item/rig_module/health_readout)
+
 	var/obj/item/clothing/head/helmet/space/rig/H = null
-	var/head_type = /obj/item/clothing/head/helmet/space/rig
+	var/obj/item/clothing/gloves/G = null
+	var/obj/item/clothing/shoes/magboots/MB = null
+	var/obj/item/weapon/tank/T = null
 	var/obj/item/weapon/cell/cell = null
+
+	var/head_type = /obj/item/clothing/head/helmet/space/rig
+	var/boots_type =  null
+	var/gloves_type = null
+	var/tank_type = null
 	var/cell_type = /obj/item/weapon/cell/high //The cell_type we're actually using
-	var/list/modules = list()
-	actions_types = list(/datum/action/item_action/toggle_rig_helmet)
+
+	var/mob/living/carbon/human/wearer
```

Comment:
This keeps an explicit reference to the wearer. Wouldn't this cause problems with soft-deletion ?

Reply:
i'm nulling them on its destroy.